src/sync.rs

```diff
@@ -16,6 +16,8 @@ pub fn sync(config: &crate::Config) -> crate::Result<()> {
     let actions = {
         let mut actions = vec![];
 
+        let tree = crate::traverse::root_tree(&std::path::PathBuf::from(&config.root))?;
+
```

Comment:
This is needed by only raw but is always calculated. Do a variable dependency analysis. Refer to `block_variable` and `variabled_proximity` in https://www.fifthtry.com/fifthtry/handbook/rust-code-guidelines/.

Reply:
This is being called outside the for loop, that's why it is kept outside.